sys/random/mersenne.c

```diff
@@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
     y ^= (y << 7) & 0x9d2c5680UL;
     y ^= (y << 15) & 0xefc60000UL;
     y ^= y >> 18;
+
+#ifdef MODULE_ENTROPY
+    return y + entropy;
```

Comment:
Hi again Kaspar,

On 18.11.2015 03:13, Kaspar Schleiser wrote:

>     hey had performed a larger study on common sources of 'entropy'
>     (typing, perturbed wireless signals, other hardware fluctuations)
>     and results were disastrous: no reliable unpredictability nowhere.
>
> I'd like to read that study.
>

I managed to get a hold of the colleague. No official report available, 
but helpful input:

  First, there is a good standardized test suite of 'randomness' for 
random number generators by NIST: 
http://csrc.nist.gov/groups/ST/toolkit/rng/documentation_software.html - 
one can use this to verify the quality of an implementation.

  Second, she said that the best pseudo-randomness at reasonable effort 
is achieved by coupling to nonlinear feedback shift register. This would 
be feasible for crypto, as well. I have not looked at specific 
implementations (and their complexity), but I guess we should keep to 
this double strategy (if needed): One very lightweight simple (not 
necessariyl crypto-safe) implementation and one that is crypto-compliant.

Thomas